applications/blackhole/src/blackhole_socket_callback.erl

```diff
@@ -14,16 +14,37 @@
 -export([open/3
         ,recv/2
         ,close/1
+
+        ,active_sessions/0, active_sessions_count/0
+        ,active_sessions/1, active_sessions_count/1
         ]).
 
+-define(SESSION_KEY(SessionId), {'session', ?MODULE, SessionId}).
+
 -type cb_return() :: {'ok', bh_context:context()}.
 
+-record(cache_info
+       ,{start_time_s = kz_time:now_s() :: kz_time:gregorian_seconds() | '_'
+        ,socket_pid :: pid() | '_'
+        ,ip :: kz_term:ne_binary()
+        }
+       ).
+
 -spec open(pid(), binary(), inet:ip_address()) -> cb_return().
-open(Pid, Id, Ipaddr) ->
-    IPBin = kz_term:to_binary(inet_parse:ntoa(Ipaddr)),
-    lager:debug("opening socket (~p) ~p, peer: ~p", [Pid, Id, IPBin]),
+open(Pid, SessionId, IPAddr) ->
+    IPBin = kz_network_utils:iptuple_to_binary(IPAddr),
+    lager:debug("opening socket (~p) ~p, peer: ~p", [Pid, SessionId, IPBin]),
 
-    Context = bh_context:set_source(bh_context:new(Pid, Id), IPBin),
+    Context = bh_context:set_source(bh_context:new(Pid, SessionId), IPBin),
+
+    kz_cache:store_local(?CACHE_NAME
+                        ,?SESSION_KEY(SessionId)
+                        ,#cache_info{'start_time_s' = kz_time:now_s()
+                                    ,'socket_pid' = self()
+                                    ,'ip' = IPBin
+                                    }
+                        ,[{'expires', ?SECONDS_IN_HOUR}]
```

Comment:
@jamesaimonetti if you expire these entries based on time, then the cache will not reflect the actual counts...